### ISSUE TITLE
Gocomics site restructured

### ISSUE THREAD

**Issue by hendersj** [webcomics/dosage#117]
Today's run showed the following results:

```
GoComics/AndyCapp> ERROR: XPath //div[contains(concat(" ", @class, " "), " gc-deck--cta-0 ")]//a[contains(concat(" ", @class, " "), "  ")] not found at URL http://www.gocomics.com/andycapp.
GoComics/ArloAndJanis> ERROR: XPath //div[contains(concat(" ", @class, " "), " gc-deck--cta-0 ")]//a[contains(concat(" ", @class, " "), "  ")] not found at URL http://www.gocomics.com/arloandjanis.
GoComics/BasicInstructions> ERROR: XPath //div[contains(concat(" ", @class, " "), " gc-deck--cta-0 ")]//a[contains(concat(" ", @class, " "), "  ")] not found at URL http://www.gocomics.com/basicinstructions.
GoComics/BC> ERROR: XPath //div[contains(concat(" ", @class, " "), " gc-deck--cta-0 ")]//a[contains(concat(" ", @class, " "), "  ")] not found at URL http://www.gocomics.com/bc.
```
It looks like the site's layout has changed somewhat - the main URL has a "read now" button that links to a URL in the format:

https://www.gocomics.com/comicname/yyyy/mm/dd

And the actual comic image is then on that page.  It looks like the image itself is contained in code that looks like this:

```
<img class="img-fluid lazyloaded" srcset="https://assets.amuniversal.com/89a72a806068013640d8005056a9545d 900w" data-srcset="https://assets.amuniversal.com/89a72a806068013640d8005056a9545d 900w" sizes="
                       (min-width: 992px) 900px,
                       (min-width: 768px) 600px,
                       (min-width: 576px) 300px,
                       900px" width="100%" alt="Arlo and Janis for Jul 9, 2018 Comic Strip" src="https://assets.amuniversal.com/89a72a806068013640d8005056a9545d">
```

With the assets.amuniversal.com link being the actual image file.

**Comment by littauer** [comment]
If I use curl or wget on https://www.gocomics.com/comicname/yyyy/mm/dd I get redirected to the main page.

**Comment by hendersj** [comment]
That's the format of the URL - for example https://www.gocomics.com/arloandjanis/2018/07/12 (this doesn't redirect for me using wget)

**Comment by knerhood** [comment]
An interesting thing (that might prove helpful) is that the comic image is referenced in a div slightly above the actual image location with a specific data value of the actual URL for the image. May be easier to grab that.

```
<div id='js-item-start' class="comic container js-comic-2573436 js-share-item js-comic-swipe bg-white border rounded comic--can-buy" data-shareable-model="FeatureItem"
data-shareable-id="2573436"
data-id="2573436"
data-feature-id="467"
data-feature-name="Arlo and Janis"
   ...
data-description="For Jul 12, 2018"
data-image="https://assets.amuniversal.com/8dcf05406068013640d8005056a9545d"

```

**Comment by littauer** [comment]
OK, I was trying with dogeatdoug, which failed until I used the dosage name of DogEatDoug, but it works now without my changing things. Strange.

Of the 50 or so GoComics I follow, the dosage name works except for RaisingDuncan (Raising-Duncan works) and InSecurity (In-Security works).

If the name you supply doesn't work, GoComics does a search to try to find what you meant (sometimes it works and sometimes it gives you something odd) and failing that sends you to the main page.

My thanks to you both for getting me back on track.

I don't understand dosage enough to create a plugin for gocomics but I hope I've contributed a little.
